packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -142,6 +142,59 @@ class _AnimatedState extends State<AnimatedWidget> {
   Widget build(BuildContext context) => widget.build(context);
 }
 
+/// Signature for a convenience function for passing around a builder for a
+/// transiton's secondary animation.
+///
+/// Typically used as the argument for [DelegatedTransition.builder].
+typedef DelegatedTransitionBuilder = Widget Function(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child);
+
+/// An exit animation for a route that can be provided to another route.
+///
+/// {@template flutter.widgets.delegatedTransition}
+/// Used for the purposes of coordinating transitions between two routes with
+/// different route transitions. When a route is added to the stack, the original
+/// topmost route will look for this transition, and if diferent from its own
+/// `DelegatedTransition`, it will use this [builder] to animate off the screen.
```

Comment:
Nit: Maybe add another paragraph with an example here?

"A page transition with a DelegatedTransition typically requires the previous page to animate in-sync with its page. For example, iOS's Modal Bottom Sheet transition moves the previous page down to appear tabbed while the new page slides in."

Eh maybe we don't want to mention MBS yet? We could link to it later when it does exist. Up to you.

Reply:
Adding the MBS explanation when that lands makes sense to me. There are two examples just outside of the template, should one or both be moved up into it?

Reply:
Or maybe just point to DelegatedTransition in a "See also" and mention that it has an example? Either way.